packages/mobile/src/components/drawers/UnfavoriteDownloadedCollection.tsx

```diff
@@ -0,0 +1,57 @@
+import { useCallback } from 'react'
+
+import type { CommonState } from '@audius/common'
+import { FavoriteSource, collectionsSocialActions } from '@audius/common'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { useDrawer } from 'app/hooks/useDrawer'
+import { removeCollectionDownload } from 'app/services/offline-downloader'
+
+import { ConfirmationDrawer } from './ConfirmationDrawer'
+
+const messages = {
+  header: 'Are You Sure?',
+  description:
+    'Unfavoriting this playlist will also remove it from your device',
+  confirm: 'Unfavorite and Remove'
+}
+
+const drawerName = 'UnfavoriteDownloadedCollection'
+const { unsaveCollection } = collectionsSocialActions
+
+export const UnfavoriteDownloadedCollectionDrawer = () => {
+  const { data } = useDrawer(drawerName)
+  const dispatch = useDispatch()
+  const { collectionId } = data
+  const collectionIdStr = collectionId.toString()
+
+  const tracksForDownload = useSelector((state: CommonState) => {
+    const collection = state.collections.entries[collectionId]
+    if (!collection) return []
+    const { tracks } = collection.metadata
+    if (!tracks) return []
+
+    const downloadReason = {
+      is_from_favorites: false,
```

Comment:
Now this is interesting... I think we actually want to remove it whether it's from favorites or not. We could have originally downloaded the collection as part of the favorites umbrella, or on the collection page. In either case, un-favoriting the collection should cause the download to be removed.

I think the solution here is actually to add two items to `tracksForDownload`: one with `is_from_favorites: true` and one that's `false`. That should still work with `removeCollectionDownload` as it does a best-effort and only removes the data once the track's remaining `downloadReasons` list is empty